test/multiplexed-session.ts

```diff
@@ -0,0 +1,289 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import {beforeEach, afterEach, describe, it} from 'mocha';
+import * as events from 'events';
+import * as sinon from 'sinon';
+import {Database} from '../src/database';
+import {Session} from '../src/session';
+import {
+  MultiplexedSession,
+  MUX_SESSION_AVAILABLE,
+  MUX_SESSION_CREATE_ERROR,
+} from '../src/multiplexed-session';
+import {Transaction} from '../src/transaction';
+import {FakeTransaction} from './session-pool';
+
+describe('MultiplexedSession', () => {
+  let multiplexedSession;
+
+  function noop() {}
+  const DATABASE = {
+    createSession: noop,
+    databaseRole: 'parent_role',
+  } as unknown as Database;
+
+  let fakeMuxSession;
+  let createSessionStub;
+  const sandbox = sinon.createSandbox();
+
+  const createSession = (name = 'id', props?): Session => {
+    props = props || {multiplexed: true};
+
+    return Object.assign(new Session(DATABASE, name), props, {
+      create: sandbox.stub().resolves(),
+      transaction: sandbox.stub().returns(new FakeTransaction()),
+    });
+  };
+
+  beforeEach(() => {
+    fakeMuxSession = createSession();
+    createSessionStub = sandbox
+      .stub(DATABASE, 'createSession')
+      .withArgs({multiplexed: true})
+      .callsFake(() => {
+        return Promise.resolve([fakeMuxSession]);
+      });
+    multiplexedSession = new MultiplexedSession(DATABASE);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('instantiation', () => {
+    it('should correctly initialize the fields', () => {
+      assert.strictEqual(multiplexedSession.database, DATABASE);
+      assert.strictEqual(multiplexedSession.refreshRate, 7);
+      assert.deepStrictEqual(multiplexedSession._multiplexedSession, null);
+      assert(multiplexedSession instanceof events.EventEmitter);
+    });
+  });
+
+  describe('createSession', () => {
+    let _createSessionStub;
+    let _maintainStub;
+
+    beforeEach(() => {
+      _maintainStub = sandbox.stub(multiplexedSession, '_maintain');
+      _createSessionStub = sandbox
+        .stub(multiplexedSession, '_createSession')
+        .resolves();
+    });
+
+    it('should create mux session', () => {
+      multiplexedSession.createSession();
+      assert.strictEqual(_createSessionStub.callCount, 1);
+    });
+
+    it('should start housekeeping', done => {
+      multiplexedSession.createSession();
+      setImmediate(() => {
+        try {
+          assert.strictEqual(_maintainStub.callCount, 1);
+          done();
+        } catch (err) {
+          done(err);
+        }
+      });
+    });
+
+    it('should propagate errors for Multiplexed Session which gets emitted', async () => {
```

Comment:
There is no assert in this test ?

Reply:
The assertion part is in the if condition
```
if (err === fakeError) {
  resolve();
}
```
if both the errors are same the promise will get resolve and test will get pass
```
else {
  reject(new Error('Unexpected error emitted'));
}
```
otherwise the promise will fail with the rejected promise

Reply:
Based on our 1:1. I have updated the test, and have used assert instead of if condition